### PR TITLE
Migrate to TF 2.4.2

### PR DESCRIPTION
**Pull request by l-bat**

### CODE REVIEW

**Comment by l-bat** [comment]
sota 260

**Comment by FyodorK** [comment]
Jenkins please stop all builds

**Comment by FyodorK** [comment]
Jenkins please retry a build

**Comment by FyodorK** [comment]
Jenkins please stop all builds

**Comment by FyodorK** [comment]
Jenkins please retry a build

**Comment by FyodorK** [comment]
Jenkins please retry a build

**Comment by FyodorK** [comment]
Jenkins please stop all builds

File: examples/tensorflow/requirements.txt [hunk 1]
```diff
@@ -1,6 +1,6 @@
 absl-py==0.10
 tensorflow_datasets==4.2.0
 tensorflow_hub
-tensorflow_addons==0.12.0
+tensorflow_addons==0.12.1
```
Comment (pfinashx) [inline examples/tensorflow/requirements.txt:4]: We have double of tensorflow_addons in [tests/tensorflow/requirements.txt](https://github.com/openvinotoolkit/nncf/blob/develop/tests/tensorflow/requirements.txt), need change it too.

**Review by pfinashx — CHANGES_REQUESTED** [review]

File: examples/tensorflow/requirements.txt (hunk 1 above)
Comment (l-bat) [reply examples/tensorflow/requirements.txt:4]: Done

**Comment by pfinashx** [comment]
Jenkins please retry a build

**Review by alexsu52 — APPROVED** [review]

**Review by pfinashx — APPROVED** [review]

**Pull request merged**